### PR TITLE
chore(docs): update docs for renku-python 1.0.0 release

### PR DESCRIPTION
**Pull request by Panaetius**
closes https://github.com/SwissDataScienceCenter/renku-python/issues/2400

### CODE REVIEW

**Comment by rokroskar** [comment]
@Panaetius should we merge this once the docs reorg is merged and once renku-python 1.0 is out?

**Comment by Panaetius** [comment]
@rokroskar would there be an issue with merging it before renku-python 1.0.0? if not, i'd say merge it

**Comment by rokroskar** [comment]
Not sure tbh - though all of the links currently live on renkulab point to the latest docs so users will see this which I guess would be confusing.

File: docs/topic-guides/provenance.rst [hunk 1]
```diff
@@ -100,18 +100,28 @@ An example execution would look something like:
...
 
 1. The command was executed.
-2. If it completed successfully, a `Common Workflow Language (CWL) <https://www.commonwl.org/>`_ tool specification was created, linking this command-line invocation to the inputs and outputs.
-3. Everything was committed to the git repository.
-
-CWL is an emerging standard for describing scientific workflows. By using this
-standard, we hope to ensure the longevity of results as well as the interoperability
-of provenance information recorded in Renku with other tools and platforms.
+2. If it completed successfully, a ``Plan`` entry was created with its ``name``
+   set to the name specified with ``--name``. This is a workflow specification
```
Comment (vigsterkr) [inline docs/topic-guides/provenance.rst:110]:
```suggestion
   set to the name specified by ``--name``. This is a workflow specification
```

File: docs/topic-guides/provenance.rst [hunk 2]
```diff
@@ -120,10 +130,11 @@ Applying the Provenance
...
 give you easy access to its benefits. Once the provenance is recorded, there are
 several ways in which it can become immediately beneficial. The most common
-usage is to **update** results when any of the input data or code
+usage is to **renku update** results when any of the input data or code
 dependencies change. By knowing exactly which results depend on a particular
 input, we can make sure to recompute only the necessary steps and not the
 entire pipeline, potentially avoiding expensive calculations in complex
-settings. For understanding the basic functionality, head to
+settings. In addition, recorded workflows can be executed independently using
+**renku workflow execute**.For understanding the basic functionality, head to
```
Comment (vigsterkr) [inline docs/topic-guides/provenance.rst:138]:
```suggestion
**renku workflow execute**. For understanding the basic functionality, head to
```

**Review by vigsterkr — COMMENTED** [review]
so very minor comments

**Comment by rokroskar** [comment]
I tried to rebase this and it was a mess - so I made a new branch `rp-2400-updates-for-1.0.0-2` that is based on current master and just cherry-picks the relevant commits. Feel free to change to that branch to make this PR a bit easier to parse. OTOH maybe github will just do the right thing when it's merged?

**Review by rokroskar — APPROVED** [review]
🚀

**Pull request merged**